### PR TITLE
[IE CORE] enable plugins & dependent libs loading using absolute path

### PR DESCRIPTION
**Pull request by mryzhov**
Initial PR - https://github.com/openvinotoolkit/openvino/pull/1098

This fix allows using plugins.xml file to specify the full path to specific plugin with it's all dependency, not to be persisted in CWD or in PATH

Those changes are also required to create self-contained wheel packages for Windows (user will not need to set PATH environments)

### CODE REVIEW

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp [hunk 1]
```diff
@@ -113,12 +116,80 @@ class SharedObjectLoader::Impl {
...
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<WCHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
+    std::basic_string<WCHAR> IncludePluginDirectory(LPCWSTR path) {
+        DWORD nBufferLength = GetDllDirectoryW(0, nullptr);
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:153]: we need to use `GetDllDirectoryA_Fnc` and so on. These functions are not available on UWP platforms, so we need dynamically decide whether can use `kernel32.dll` or not

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp [hunk 2]
```diff
@@ -113,12 +116,80 @@ class SharedObjectLoader::Impl {
...
+        return original;
+    }
+
 #ifdef ENABLE_UNICODE_PATH_SUPPORT
+    static const wchar_t* FindLastPathSeparator(LPCWSTR path) {
+        const wchar_t* const last_sep = wcsrchr(path, kPathSeparator);
+        return last_sep;
+    }
+
+    std::basic_string<WCHAR> GetDirname(LPCWSTR path) {
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:142]: `static`?

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp [hunk 3]
```diff
@@ -113,12 +116,80 @@ class SharedObjectLoader::Impl {
...
+        auto pos = FindLastPathSeparator(path);
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<WCHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
+    std::basic_string<WCHAR> IncludePluginDirectory(LPCWSTR path) {
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:152]: why not `std::wstring`?

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp (hunk 1 above)
Comment (mryzhov) [reply inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp]: modified, but need to define the right condition, for now, I changed it to !WINAPI_PARTITION_SYSTEM

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp (hunk 2 above)
Comment (mryzhov) [reply inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp]: fixed

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp (hunk 3 above)
Comment (mryzhov) [reply inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp]: it was in initial PR, changed to string/wstring

File: inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp [hunk 4]
```diff
@@ -129,8 +213,12 @@ class SharedObjectLoader::Impl {
 
     explicit Impl(const char* pluginName) {
         ExcludeCurrentDirectoryA();
+        LoadPluginFromDirectoryA(pluginName);
+
+        if (!shared_object) {
```
Comment (ilya-lavrenov) [inline inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:218]: will we have situations when `shared_object` is not loaded in `LoadPluginFromDirectoryA`?

Comment (mryzhov) [reply inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:218]: Theoretically  yes: 
#if !WINAPI_PARTITION_SYSTEM
        LoadSymbols();
        if (IEGetDllDirectoryA)

Comment (ilya-lavrenov) [reply inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp:218]: ok. I see

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**